teuthology/suite/run.py

```diff
@@ -206,7 +207,14 @@ def choose_teuthology_branch(self):
                     "branch {0} not in teuthology.git; will use master for"
                     " teuthology".format(self.args.ceph_branch))
                 teuthology_branch = 'master'
-        log.info("teuthology branch: %s", teuthology_branch)
+        teuthology_hash = util.git_ls_remote(
```

Comment:
Have you tested this test with 'refs' branches?

Reply:
No. How do I do that?

Reply:
Deploy the teuthology for this PR and try to schedule a build with for example this PR ref branch.

Reply:
If by "deploy teuthology for this PR" you mean to run bootstrap on teuthology.front, then I can do so.